content/components/uart.md

```diff
@@ -65,10 +65,17 @@ uart:
 - **rx_pin** (*Optional*, [Pin](#config-pin)): The pin to receive data on from the ESP's perspective. Use the full pin
   schema and set `inverted: true` to invert logic levels. Not supported by host platform.
 
+- **flow_control_pin** (*Optional*, [Pin](#config-pin)): ESP32 only. The pin used to for hardware RS485 flow control.
+  When using this setting, half-duplex mode is enabled. Use the full pin schema and set `inverted: true` to invert logic levels.
+
 - **port** (*Optional*, string): Host platform only. Unix style name of the port to use.
 - **rx_buffer_size** (*Optional*, int): The size of the buffer used for receiving UART messages. Increase if you use an
   integration that needs to read big payloads from UART. Defaults to `256`.
 
+- **rx_full_threshold** (*Optional*, int): ESP32 only. After this many bytes have been received, the data becomes available for processing.
+  Defaults calculate during compilation to be approximately 10ms (about 8 bytes at 9600 baud, 114 bytes at 115200 baud).
+  
+- **rx_timeout** (*Optional*, int): ESP32 only. After this amount time of no data been received (in bytes-time), the data becomes available for processing. Defaults to ``2``.
```

Comment:
```suggestion
- **rx_timeout** (*Optional*, int): ESP32 only. This value specifies a number of bytes used to determine the duration of the timeout. The duration of the timeout is based on the amount of time it would take to transmit/receive this number of bytes. In other words, for a given value and baud rate, doubling the baud rate will halve the timeout value. After the timeout has elapsed, the data becomes available for processing. Defaults to ``2``.
```

Reply:
Great, so this is a clear indication that I didn't understand it, either. 😅

> It's not "transmit/receive", only receive.

Easy to fix, although is the transmit speed not the same as the receive speed? I suppose it doesn't technically _have_ to be, but I don't think we support different speeds for Tx and Rx.

> And "doubling the baud rate will halve the timeout value" could be misread, the timeout value does not change, the time changes.

Does the amount of time not determine the duration of the timeout? Maybe I'm misunderstanding what's going on here.

Reply:
Maybe this?:

```suggestion
- **rx_timeout** (*Optional*, int): ESP32 only. This value specifies a number of bytes used to determine the duration of the timeout. The duration of the timeout is based on the amount of time it would take to receive this number of bytes. In other words, for a given value and baud rate, doubling the baud rate will halve the duration of the timeout. After the timeout has elapsed, the data becomes available for processing. Defaults to ``2``.
```